daemon/exec_linux.go

```diff
@@ -90,7 +91,13 @@ func (daemon *Daemon) execSetPlatformOpt(ctx context.Context, daemonCfg *config.
 			}
 		}
 		p.ApparmorProfile = appArmorProfile
+	} else {
+		// If AppArmor is not supported but a profile was specified, return an error
+		if ec.Container.AppArmorProfile != "" {
+			return errors.New("AppArmor is not supported on this host, but the profile '" + ec.Container.AppArmorProfile + "' was specified")
+		}
 	}
+
```

Comment:
I wonder if we could actually hit this case; this would mean that a container was created (and is running) with the profile, but `docker exec` into the container would fail 🤔

Reply:
it hasn't happened to me so far, but I doubt having this code would cause any harm.